### ISSUE TITLE
It's been a fun ride

### ISSUE THREAD

**Issue by loopuleasa** [loopuleasa/Dota2-TheCore-Config-Engine#161]
With https://github.com/loopuleasa/Dota2-TheCore-Config-Engine/issues/159 update coming along, it seems that the bulk of the Dota2 Core Engine won't work any longer and it saddens me to see it go.

I want to thank everyone that contributed to this project. This scripting engine was a very fun learning experienced that thought me many concepts along the way of my programming career. It even helped me get a job working as a C developer, since it was such a good lightsource on my portfolio. Learning how to use git, how to collaborate on code works, how to delegate work, how to manage my time and meet deadlines when patches arrive, how to handle marketing aspects, how to develop systems on top of available systems, how to design architectures for those systems. All of these, I've learned while pursuing a need that shot through my mind on a sunny day when I was butthurt over the options Valve gave us through the in-game menu, and through the fascinating possibilities a simple console provided.

This engine, together with the people developing it, was always against macros/hacks/cheats and thus we were careful to make every single keystroke available within to only launch one in-game action. You perform one or more actions, only one action comes out the pipe at the other end in the Dota2 client. That was the concept, and we stuck to that. The only difference was how the layout of the available commands should be spread, in a smart manner, an efficient one that was user friendly for all the players out there. Maximizing the space you have on the keyboard while providing you as much control was our end goal and I am proud to say that everyone that contributed to this project, together, we managed to achieve that.

I am saddened to see so many people being forced to switch to other layouts, limited to the in-game menu, after becoming so familiarized with my super compact keyboard layout. I am not sure how I will play dota now. It's like drinking the best beer in the world for two years straight, then having to go and drink piss-water just to maintain my alcoholism.

I'd rather just quit drinking. Or playing the game, in this metaphor.

I will try to see if there are any workarounds once the patch hits ( if it didn't already ) but I am sad to see that there will be limitations and that Valve decided to strip us of the beautiful control we've been given up until now, just because some people found ways to abuse the autoexec.

Thanks for everything, everyone.
Loop.

**Comment by dodekeract** [comment]
I'm pretty sure we have to either re-learn everything (fuck youj valve) or use autohotkey/etc, which feels like cheating.

**Comment by AetherV** [comment]
I have a concept that may work but will require a lot of testing to make it work. It also might not be able to fulfill all the old stuff, but it might work for key modifiers at a bare minimum

**Comment by loopuleasa** [comment]
Thread is up

https://www.reddit.com/r/DotA2/comments/4kzld1/a_plea_to_valve_please_dont_kill_the_honest/

**Comment by dodekeract** [comment]
@loopuleasa I think you should add a small section about the possible solution, which is to prevent multiple item/ability uses per autoexec-tick.

**Comment by silencer666** [comment]
Thanks Loop. Much love.

**Comment by thth** [comment]
@AetherV Could you detail your workaround?

**Comment by AetherV** [comment]
@thth Wrong thread but way I did it
Only tested it in a demo game so far (sv_cheats turned manually off)
In lobby game with cheats for levels it works, in lobby without anything I can't even execute a simple say_student script.


[poc.cfg.txt](https://github.com/loopuleasa/Dota2-TheCore-Config-Engine/files/282623/poc.cfg.txt)
[autoexec.cfg.txt](https://github.com/loopuleasa/Dota2-TheCore-Config-Engine/files/282624/autoexec.cfg.txt)

**Comment by crimsonvspurple** [comment]
@AetherV, as far as I can understand, that's the "fix"; nothing to do with "your end."

**Comment by AetherV** [comment]
Wait, so you telling me. I am not allowed to do `bind x say_student hello`

**Comment by thth** [comment]
@AetherV ``sv_cheats 0`` in a demo game doesn't seem to actually turn cheats off; as far as I can tell ``exec`` unfortunately isn't usable in an actual game

A simple ``bind x say_student hello`` should still work though

**Comment by AetherV** [comment]
This does not work for me
[new2.cfg.txt](https://github.com/loopuleasa/Dota2-TheCore-Config-Engine/files/282672/new2.cfg.txt)

So where are our keybindings than saved? Because dota has got to save our hotkey list somewhere.

**Comment by thth** [comment]
@AetherV maybe your ``x`` key is overriden in the in-game hotkey config or maybe you have your line behind an ``exec`` in your autoexec; just inputting ``bind x say_student hello2`` in the console works.

**Comment by AetherV** [comment]
Strange. Could be. So am I mistaken in thinking that the exec command works nowhere but in the main menu?

**Comment by thth** [comment]
I haven't done extensive testing but I'd imagine if you're in a server with cheats off ``exec``, ``alias``, and anything after a ``;`` won't work

**Comment by AetherV** [comment]
yeah great, might be able to fiddle around with a couple of files directly related to key binding,but am not hopeful. Guess Valve truly did do the nuclear option, kinda shit that we got caught in the crossfire. Like seriously though, if they would sort out their settings, than there would not be a need for configs like this, but I guess that is Valve. Kinda glad I didn't buy a compendium.

**Comment by jaccarmac** [comment]
It has been a fun ride, all. Thanks for everyone's hard work, you guys made it a lot easier to tweak my binds and use them across PCs and operating systems. Comment posted on the reddit thread though I'm afraid the anti-scripting crowd is drowning out useful discussion in most places :-/.

**Comment by abhinav91690** [comment]
Thanks a lot loop and devs for the this. Looking at the current Reddit threads' status, popular opinion seems to be against this , Which valve will follow for sure. I hope they will give us enough options in the game to have majority of our binds working.

**Comment by dodekeract** [comment]
@abhinav91690 I made an issue for possible workarounds. #162 is what you should follow for "enough options".

**Comment by Karrandalf** [comment]
For those out of the loop, we have a simple, single AutoHotKey solution now. This is all it takes:

    ~*Space::ControlSend, , {LWin Down}, Dota 2 
    *Space Up::Send {LWin Up}

https://github.com/Karrandalf/D2SpaceModifier
